src/app/components/Heading/index.tsx

```diff
@@ -0,0 +1,47 @@
+import classNames from 'classnames';
+import React from 'react';
+import styles from './index.module.scss';
+
+export enum HeadingStyles {
+  'h1' = 'h1',
+  'h2' = 'h2',
+  'h3' = 'h3',
+  'h4' = 'h4',
+  'h5' = 'h5',
+  'h6' = 'h6',
+  'label' = 'label',
+}
+
+type IHeadingProps = {
+  style?: HeadingStyles;
+  children: React.ReactNode;
+  className?: string;
+  id?: string;
+};
+
+const Heading: (tag: string) => React.FC<IHeadingProps> = tag => ({
+  style,
+  className,
+  id,
+  children,
+}) => {
+  return React.createElement(
+    tag,
+    {
+      className: classNames(
+        style ? styles[style.toString()] : styles[tag],
+        className,
+      ),
+      id,
+    },
+    children,
+  );
+};
+
+export const H1 = Heading('h1');
```

Comment:
Use `HeadingStyles` enum instead of string literals. You can maybe rename the enum to `HeadingTypes`, it would probably make more sense since you're going to use it here.